presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/MapColumnValidator.java

```diff
@@ -44,16 +46,28 @@ public List<SingleColumn> generateChecksumColumns(Column column)
 
         Expression checksum = functionCall("checksum", column.getExpression());
         Expression keysChecksum = generateArrayChecksum(functionCall("map_keys", column.getExpression()), new ArrayType(keyType));
-        Expression valuesChecksum = generateArrayChecksum(functionCall("map_values", column.getExpression()), new ArrayType(valueType));
         Expression mapCardinalityChecksum = functionCall("checksum", functionCall("cardinality", column.getExpression()));
         Expression mapCardinalitySum = new CoalesceExpression(
                 functionCall("sum", functionCall("cardinality", column.getExpression())),
                 new LongLiteral("0"));
 
+        // We need values checksum in one case only: when key is a floating point type and value is not.
+        // In such case, when both checksum and the key checksum are bad, we cannot tell if the values match or not.
```

Comment:
by bad, do you mean that the control and test checksum do not match?

So the situation is:
"checksum" is a checksum of the whole map.
"keysChecksum" is a checksum of just the keys.

When we have floating point types, we expect that the checksums may not match due to non-determinism in floating point types, so we may want to autoresolve the mismatch failures.  However, if the values are not floating point types, then any mismatches there could be real errors, so we first need to checksum the values and make sure those match.

Reply:
@rschlussel Yes, like you described.
Here is the sheet with all possible combinations: [table](https://docs.google.com/spreadsheets/d/16b6WMKi_8WPy0eWJRjVm-FT59kUtLCbfuCkjAzUE3j0/edit?usp=sharing).
We need values checksum in two of them, where '?' is.